src/benchmark/measurement.ts

```diff
@@ -1,62 +1,92 @@
+
+        Measurement.executeJS(func[0], args, expectedResult, warmupRounds);
+        const jsTimes = Measurement.executeJS(func[0], args, expectedResult, measureRounds);
+
+        const wasmArgs = args.slice();
+        wasmArgs.unshift(func[0].name);
+
+        const wasmAlgorithm = Transpiler.transpile(func.join(''));
+        Measurement.executeWasm(wasmAlgorithm, wasmArgs, expectedResult, warmupRounds);
+        const wasmTimes = Measurement.executeWasm(wasmAlgorithm, wasmArgs, expectedResult, measureRounds);
+
+        return [jsTimes, wasmTimes];
+    }
+
     // noinspection TsLint
-    private static executeAlgorithm(algorithm: Function,
-                                    args: any[],
-                                    expectedResult: number,
-                                    rounds: number): number[] {
-        const times: number[] = [];
+    private static executeJS(algorithm: Function,
+                             args: any[],
+                             expectedResult: any,
+                             rounds: number): Timing[] {
+        const times: Timing[] = [];
 
         for (let i = 0; i < rounds; i++) {
-            const startTime: number = Date.now();
+            const start = performance.now();
             const result = algorithm(...args);
-            const endTime: number = Date.now();
-            times.push(endTime - startTime);
+            const executionTime = performance.now() - start;
 
-            // noinspection TsLint because of boolean handling in wasm
-            if (result != expectedResult) {
-                console.log(`Result was wrong, expected=${expectedResult}, actual=${result}`);
-            }
+            Measurement.assert(result, expectedResult);
+
+            times.push({
+                compilationTime: 0,
+                executionTime,
+                importTime: 0,
+            });
         }
 
         return times;
     }
 
-    private readonly warmupRounds: number;
-    private readonly measureRounds: number;
-
-    constructor(warmupRounds: number, measureRounds: number) {
-        this.warmupRounds = warmupRounds;
-        this.measureRounds = measureRounds;
-    }
-
     // noinspection TsLint
-    public measure(algorithm: Algorithm): [number[], number[]] {
-        const func = algorithm.func;
-        const args = algorithm.arguments;
-        const expectedResult = algorithm.expectedResult;
+    private static executeWasm(algorithm: Function,
+                             args: any[],
+                             expectedResult: any,
+                             rounds: number): Timing[] {
+        const times: Timing[] = [];
 
-        Measurement.executeAlgorithm(func[0], args, expectedResult, this.warmupRounds);
-        const jsTimes = Measurement.executeAlgorithm(func[0], args, expectedResult, this.measureRounds);
+        for (let i = 0; i < rounds; i++) {
+            const result = algorithm(...args);
```

Comment:
We need to work with a deep copy of the args array. Otherwise object parameters do not always have the initial values.